tools/wifi_scanner.py

```diff
@@ -0,0 +1,177 @@
+import serial
+import time
+import json
+from typing import List, Dict
+
+class ESPWiFiScanner:
+    def __init__(self, port: str, baudrate: int = 115200):
+        self.port = port
+        self.baudrate = baudrate
+        self.serial = None
+
+    def connect(self) -> bool:
+        try:
+            self.serial = serial.Serial(
+                port=self.port,
+                baudrate=self.baudrate,
+                timeout=1
+            )
+            return True
+        except serial.SerialException as e:
+            print(f"Error connecting to ESP32: {e}")
+            return False
+
+    def scan_networks(self, timeout_seconds: int = 30) -> List[Dict]:
+        if not self.serial:
+            print("Not connected to ESP32")
+            return []
+
+        self.serial.reset_input_buffer()
+        
+        command = '{"commands":[{"command":"scan_networks","data":{}}]}\n'
+        self.serial.write(command.encode())
+        
+        timeout_start = time.time()
+        response_buffer = ""
+        
+        while time.time() - timeout_start < timeout_seconds:
+            if self.serial.in_waiting:
+                data = self.serial.read(self.serial.in_waiting).decode('utf-8', errors='ignore')
+                response_buffer += data
+                
+                # Look for WiFi networks JSON directly (new format)
+                # The scan command now outputs JSON directly followed by command result
+                if '{"networks":[' in response_buffer:
+                    import re
+                    
+                    # Look for the networks JSON pattern that appears first
+                    networks_pattern = r'\{"networks":\[.*?\]\}'
```

Comment:
I kinda hate that we have to rely on string parsing / regex to parse the serial info that was supposed to be just json. Purely to make things simpler for tools like these. Is our output still somewhat broken?